app/presenters/sufia/file_set_presenter.rb

```diff
@@ -6,6 +6,14 @@ class FileSetPresenter < ::CurationConcerns::FileSetPresenter
     delegate :depositor, :keyword, :date_created, :date_modified, :itemtype,
              to: :solr_document
 
+    def page_title
+      title.first
+    end
+
+    def link_name
+      current_ability.can?(:read, id) ? page_title : 'File'
+    end
+
```

Comment:
Yeah, I'm in agreement with @mjgiarlo. Mostly I go with unit tests because they're just faster. Although there's already a feature test with the necessary setup, I think a unit test is better because it'll show exactly where this is occurring, which I'm still not clear on presently. You could do something like:

https://github.com/projecthydra/sufia/blob/master/spec/views/curation_concerns/base/_items.html.erb_spec.rb

But in the partial where the file's title is being rendered.